### DISCUSSION TITLE
Download dependencies using Git

### DISCUSSION THREAD

**Discussion by oraqlle** [cabinpkg/cabin#757] [Q&A] [answered]
Is it currently possible to install dependencies using Git from GitHub, GitLab etc?

**Comment by ken-matsui** [comment, marked as answer]
@oraqlle Good suggestion! I've been dedicating a lot of time to creating a Poac registry, but it's quite a complicated process. In the meantime, enabling support for Git dependencies would be a helpful and feasible solution. I'll make this a priority and work on it during my next free moment.

**Reply by oraqlle** [reply]
It could also increase adoption as other developers can easily configure their existing libraries, packages and projects to comply with Poac and use them without having to go through the registry while it is being developed. Less waiting for common packages to port to Poac and more people and create ports/branches on their own that support Poac.

**Reply by ken-matsui** [reply]
Agreed. @wx257osn2, are you interested in working on this?

**Reply by wx257osn2** [reply]
Actually yes, but currently I have no concrete progress. I implemented [`registries` feature](https://github.com/poac-dev/poac/blob/ea7c2fc343787c46c251981378f222f273f288d9/lib/core/resolver.cc#L206-L234) with git support in mind for the future, but I want ["Build packages separated by headers & implementations"](https://github.com/poac-dev/poac#roadmap) feature before implementing that to support such libraries. And I also have less free time, so... interested but doubtful about I can do.

**Reply by oraqlle** [reply]
Understandable and I agree that supporting more general packages will be better for Poac to be more feature complete and with the ability to use Conan as a registry this already offers a great deal of convenience to users and library developers.

However, this is an interesting project and one that seems to have more concrete goals and with a good idea of what are simple and effective package manager and build system for C++ can look like; albeit very similar to Cargo, which is not a bad thing a lot of the time. So if you want help continuing to develop Poac I'd be more than happy to help when I can make time be it this feature set or another.

**Comment by ken-matsui** [comment]
This was implemented in https://github.com/poac-dev/poac/pull/799.
